doc-examples/coherence-example-java/src/test/java/io/micronaut/coherence/data/AbstractDataTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.coherence.data;
+
+import com.tangosol.net.NamedMap;
+import com.tangosol.util.UUID;
+import io.micronaut.coherence.annotation.SessionName;
+import io.micronaut.coherence.data.model.Author;
+import io.micronaut.coherence.data.model.Book;
+import io.micronaut.coherence.data.util.EventRecorder;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Base test class using {@link Book}s for functional validation.
+ */
+public abstract class AbstractDataTest {
```

Comment:
Copied tests from micronaut-coherence-data
Works ok in 4.9.x. but some repos are not compiling in 4.10.x